tests/submissions/test_examples.py

```diff
@@ -0,0 +1,169 @@
+from hera.workflows.models import Workflow as ModelWorkflow
+from hera.workflows.service import WorkflowsService
+from hera.workflows.workflow import Workflow
+from hera.workflows.workflow_template import WorkflowTemplate
+from tests.test_examples import CI_MODE, _get_examples
+
+# Skip long-running and unsupported examples
+TIME_LIMIT_SECONDS = 120
+SKIP_LINT_EXAMPLES = {
+    "new_decorators_auto_template_refs",  # Uses TemplateRefs to non-existent WorkflowTemplates
+    "parquet_pandas",
+}
+SKIP_SUBMISSION_EXAMPLES = SKIP_LINT_EXAMPLES.union(
+    {
+        "artifact_loaders",
+        "create_volume_for_workflow",
+        "conditional_on_task_status",
+        "dag_input_output",
+        "recursive_dag",
+        "new_dag_decorator_artifacts",
+        "new_dag_decorator_inner_dag",
+        "new_dag_decorator_params",
+        "new_decorators_auto_template_refs",
+        "new_decorators_basic_script",
+        "new_steps_decorator_with_parallel_steps",
+        "script_runner_io",
+        "template_sets",
+        "custom_serialiser",
+        "runner_artifacts",
+        "typed_script_input_output",
+        "dynamic_fanout_container",
+        "container_set_with_env",
+        "cron_workflow_stop_strategy",
+        "data",
+        "env",
+        "env_from",
+        "global_config",
+        "http",
+        "init_containers_with_env",
+        "resource_flags",
+        "retry_workflow",
+        "suspend",
+        "suspend_input_duration",
+        "user_container",
+        "volume_mount",
+        "volume_mounts_nfs",
+        "volume_mounts_wt",
+        "workflow_with_global_params",
+        "recursive_steps",
+        "dask",
+        "fine_tune_llama",
+        "map_reduce",
+        "spacy_inference_pipeline",
+        "spark",
+        "testing_templates_and_workflows",
+        "wandb_ml_monitoring",
+        "workflow_of_workflows",
+    }
+)
+
+
+@pytest.fixture
+def workflows_service():
+    workflows_service = WorkflowsService(
+        host="https://localhost:2746",
+        namespace="argo",
+        verify_ssl=False,
+    )
+    yield workflows_service
+
+
+@pytest.mark.on_cluster
+@pytest.mark.parametrize(
+    "module_name",
+    [
+        pytest.param(
+            module_name,
+            marks=pytest.mark.skipif(
+                bool(CI_MODE) or filename in SKIP_SUBMISSION_EXAMPLES, reason="Run all examples on local install only"
+            ),
+            id=filename,
+        )
+        for _, module_name, filename in _get_examples()
+    ],
+)
+def test_submission(
+    module_name,
+    global_config_fixture,
+    workflows_service: WorkflowsService,
+):
+    global_config_fixture.host = "https://localhost:2746"
+    workflow = importlib.import_module(module_name).w
+
+    # Skip anything not a Workflow
+    if isinstance(workflow, (WorkflowTemplate, ClusterWorkflowTemplate, CronWorkflow)):
+        assert False, "Add non-workflow example to 'SKIP_SUBMISSION_EXAMPLES' set"
```

Comment:
note: [pytest.fail](https://docs.pytest.org/en/stable/reference/reference.html#pytest.fail) is also an option